### ISSUE TITLE
Start Times for Carrier Aircraft Stay at 00:00:00 Regardless of TOT (and ASAP Gives TOT of 00:00:00)

### ISSUE THREAD

**Issue by jeetsukumaran** [dcs-liberation/dcs_liberation#819]
Before filing, please search the issue tracker to see if the issue has already been reported.

**Describe the bug**

Start Times for Carrier Aircraft Stay at 00:00:00 Regardless of TOT (and ASAP Gives TOT of 00:00:00). 

I had AI aircraft default to Runway Start, so presumably OCA/Aircraft is broken and is not expected to work. However, this issue also applies to OCA/Runway, which should not be affected? Furthermore, when adding flights from landbases, this issue is not there. Only affects carrier aircraft (and regardless of their start type).

**To Reproduce**
0. Aircraft default to Runway Start
1. Create an Attack Airfield mission
2. Add any type of flight; SEAD, Escort, OCA/Airfield, etc. from a carrier.
3. See start times not affected by TOT (remain at 00:00:00) and ASAP rresults in TOT of 00:00:00.
4. Issue persists even if indivvidual flights change start type
5. Issue is not there when using land based aircraft

**Expected behavior**

Start times change based on TOT. ASAP gives reasonable TOT.

**Notes**

- Found same bug with carrier aircraft packages to frontlines (any mission)

**Version information (please complete the following information):**
 - DCS Liberation [e.g. 2.3.1]: Preview Build 1114
![Capture](https://user-images.githubusercontent.com/26183/106379685-87218d80-6362-11eb-861b-8bf319f08662.PNG)
![Capture2](https://user-images.githubusercontent.com/26183/106379686-87ba2400-6362-11eb-9347-1dd873a0d056.PNG)

**Comment by DanAlbert** [comment]
Start time 0 regardless of TOT usually means a custom flight plan (guessing that's not the case), or an error in the log typically caused by no actual flight plan being generated. Anything in the log?

**Comment by jeetsukumaran** [comment]
For the OCA/Runway:

```
Traceback (most recent call last):
  File "qt_ui\windows\mission\QPackageDialog.py", line 182, in add_flight
  File "gen\flights\flightplan.py", line 762, in populate_flight_plan
  File "gen\flights\flightplan.py", line 834, in regenerate_package_waypoints
  File "gen\flights\flightplan.py", line 880, in preferred_join_point
  File "gen\flights\flightplan.py", line 1524, in package_airfield
RuntimeError: Could not find any airfield assigned to this package
2021-01-31 01:04:18,520 :: WARNING :: Cannot estimate TOT for [OCA/Runway] 2 x F/A-18C Hornet (Lot 20)
Traceback (most recent call last):
  File "qt_ui\windows\mission\QPackageDialog.py", line 182, in add_flight
  File "gen\flights\flightplan.py", line 762, in populate_flight_plan
  File "gen\flights\flightplan.py", line 834, in regenerate_package_waypoints
  File "gen\flights\flightplan.py", line 880, in preferred_join_point
  File "gen\flights\flightplan.py", line 1524, in package_airfield
RuntimeError: Could not find any airfield assigned to this package
```

```
2021-01-31 01:04:50,934 :: WARNING :: Cannot estimate TOT for [OCA/Runway] 2 x F/A-18C Hornet (Lot 20)
2021-01-31 01:04:50,934 :: WARNING :: Cannot estimate TOT for [Escort] 2 x F/A-18C Hornet (Lot 20)
2021-01-31 01:04:55,405 :: WARNING :: Cannot estimate TOT for [OCA/Runway] 2 x F/A-18C Hornet (Lot 20)
2021-01-31 01:04:55,405 :: WARNING :: Cannot estimate TOT for [Escort] 2 x F/A-18C Hornet (Lot 20)
2021-01-31 01:04:59,374 :: WARNING :: Cannot estimate TOT for [OCA/Runway] 2 x F/A-18C Hornet (Lot 20)
2021-01-31 01:04:59,374 :: WARNING :: Cannot estimate TOT for [Escort] 2 x F/A-18C Hornet (Lot 20)
2021-01-31 01:06:20,767 :: WARNING :: Cannot estimate TOT for [OCA/Runway] 2 x F/A-18C Hornet (Lot 20)
2021-01-31 01:06:20,768 :: WARNING :: Cannot estimate TOT for [Escort] 2 x F/A-18C Hornet (Lot 20)
```
[liberation.log](https://github.com/Khopa/dcs_liberation/files/5899467/liberation.log)

**Comment by DanAlbert** [comment]
That'll do it. For some reason it's trying to create the flight plan before the flight has been assigned an airfield, which I'd have thought was impossible. Will need to dig a bit. As long as those repro steps are adequate shouldn't be too hard. Could you attach the save in case it's dependent on campaign/departure/target/etc?

**Comment by DanAlbert** [comment]
There was a prior report of this that did include a save game but it didn't repro for me: https://github.com/Khopa/dcs_liberation/issues/790

**Comment by jeetsukumaran** [comment]
Here you go,

Thanks for the Runway Start option BTW ... looking forward to trying it out.
[Current.zip](https://github.com/Khopa/dcs_liberation/files/5899477/Current.zip)

**Comment by DanAlbert** [comment]
Works for me. I did have to delete a bunch of your packages and recreate them, so I have no idea how you did it exactly. If you can upload a save that's ready for me to click a button and see a crash that'd remove the guesswork.

**Comment by DanAlbert** [comment]
Looking more closely at the code that's throwing the error, it's trying to find the closest airfield to the target that has a flight of the package assigned, but has confirmed that the package is non-empty. Airfields are not checked, it seems, if the runway was non-operational when the cache is initially primed. The cache is not a part of the save game, but is reset on load, so simply me loading your save is enough to fix the problem. Looks promising.

**Comment by DanAlbert** [comment]
Fixed. Creating a new game was a required step for repro.

1. Create new game with a faction that has a carrier (I used black sea with the default factions and settings)
2. Buy aircraft at the CVN (I bought 2 hornets)
3. Delete CAP packages
4. Create OCA/Runway mission at Gudata using hornets off the CVN

**Comment by jeetsukumaran** [comment]
Glad you figured it out. One thing weird about the save file is that I loaded it now and it does not show all the packages I had set up. There were four or five more (including the carrier launched ones). I typically save after creating a package, so in principle there should have been 4-5 saves. But this morning I loaded the save file, and it's like those 4-5 saves did not happen. Not sure if I messed up or something else going on. Anyway just to explain the discrepancy.
